src/nvim/ex_cmds.lua

```diff
@@ -2678,18 +2678,18 @@ return {
     func='ex_ni',
   },
   {
-    command='tearoff',
-    flags=bit.bor(NEEDARG, EXTRA, TRLBAR, NOTRLCOM, CMDWIN),
-    addr_type=ADDR_LINES,
-    func='ex_tearoff',
-  },
-  {
     command='terminal',
     flags=bit.bor(BANG, FILES, CMDWIN),
     addr_type=ADDR_LINES,
     func='ex_terminal',
   },
   {
+    command='tearoff',
+    flags=bit.bor(NEEDARG, EXTRA, TRLBAR, NOTRLCOM, CMDWIN),
+    addr_type=ADDR_LINES,
+    func='ex_tearoff',
+  },
```

Comment:
This rain must be kept in alphabetical order. I'm really not sure it makes sense to swap these two elements just so that we can give the `:te` prefix to `:terminal`. 

If I'm not mistaken, Vim uses a binary search to find commands, so the fact that this works seems like just a quirk of one being expanded before the other, and I wonder if this fix might be undone as we add more commands.